### PR TITLE
Intro paras

### PR DESCRIPTION
**Pull request by hillaryj**
Correctly applies intro paragraphs to references.
Adds tests to check for correct formatting of callouts and references in intro
paragraphs in sections and appendixSections.

### CODE REVIEW

File: regulation/tree.py [hunk 1]
```diff
@@ -1085,3 +1074,41 @@ def build_analysis_dict(child_elm):
...
+    :rtype: boolean
+    """
+    if item.find('{eregs}title') is None and item.get('marker') == '':
+        # Note: item[0] is always a <content> tag - check that element's children
+        if len(filter(lambda child: child.tag in NON_PARA_SUBELEMENT, item[0].getchildren())) == 0:
+            return True
+
+    return False
+
+def wants_intro_text(element):
```
Comment (willbarton) [inline regulation/tree.py:1097]: I love this function name.

File: regulation/tree.py [hunk 2]
```diff
@@ -244,10 +239,10 @@ def build_paragraph_marker_layer(root):
         transformation into JSON for use with the eRegs frontend.
     :rtype: :class:`collections.OrderedDict`:
     """
-    parapgraphs = root.findall('.//{eregs}paragraph') # + root.findall('.//{eregs}interpParagraph')
+    paragraphs = root.findall('.//{eregs}paragraph') # + root.findall('.//{eregs}interpParagraph')
     paragraph_dict = OrderedDict()
 
-    for paragraph in parapgraphs:
+    for paragraph in paragraphs:
```
Comment (willbarton) [inline regulation/tree.py:245]: :+1:

**Comment by willbarton** [comment]
:+1: This looks fantastic. My only major comment would be: shouldn't the tests be in `tests/regulation_tree_tests.py`? 

For regulations-xml-parser, we followed a convention from the other eRegs repos of putting the tests in `tests/` and naming the files for the modules, with underscores for packages. So, `regulations.tree` tests are in `regulations_tree_tests.py`.

I _think_ reg_xml_parser_tests was intended for tests for `reg_xml_parser.py`. We should probably consider a `regml_tests.py` to correspond to `regml.py`.

**Comment by hillaryj** [comment]
Yup, that's probably a better place for the tests to live. I've moved them now!

**Comment by willbarton** [comment]
giphy-5

**Pull request merged**